src/java/org/apache/cassandra/cql3/statements/CQL3CasRequest.java

```diff
@@ -149,7 +149,7 @@ public void addConditions(Clustering<?> clustering, Collection<ColumnCondition>
         }
         else if (!(condition instanceof ColumnsConditions))
         {
-            throw new InvalidRequestException("Cannot mix IF conditions and IF NOT EXISTS for the same row");
+            throw new InvalidRequestException("Cannot mix IF conditions and " + ((ToCQL) condition).toCQL() + " for the same row");
```

Comment:
this was an error msg bug, `IF EXISTS` was what the user provided but we told them `IF NOT EXISTS`.